atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/ConflictDetectionManager.java

```diff
@@ -39,7 +39,11 @@ public class ConflictDetectionManager {
      *  (This has always been the behavior of this class; I'm simply calling it out)
      */
     public ConflictDetectionManager(CacheLoader<TableReference, ConflictHandler> loader) {
-        this.cache = Caffeine.newBuilder().maximumSize(100_000).build(loader);
+        this.cache = Caffeine.newBuilder()
+                .initialCapacity(256)
+                .maximumSize(100_000)
+                .buildAsync(loader)
+                .synchronous();
```

Comment:
Can you share the original JFR or other traces indicating we're seeing contention here? I'm a little surprised as I would expect most writes to these caches occur via the `warmCacheWith` via a single thread for that `ConflictDetectionManager` instance, those writes are all serialized so there should not contend.

If we want the async loading, are we concerned that not providing an executor will cause saturation of the [default executor `ForkJoinPool.commonPool()`](https://github.com/ben-manes/caffeine/wiki/Population#asynchronous-manual)?

Reply:
Hey provided the JFR in the slack thread as it could be unsafe. Will tag you there too.

Reply:
<img width="1029" alt="image" src="https://github.com/palantir/atlasdb/assets/518196/263a9c08-0907-4094-99ee-95456c9ada3b">

Reply:
@schlosna @priyanshukm @ergo14 - what's the status on this one? Do you believe it valuable to continue reviewing, or is the issue solved some other way?

Reply:
Main thing I had in mind was to check if things improved after https://github.com/palantir/atlasdb/pull/7224

Reply:
Haven't seen any issues recently after the above PR was merged. Closing this PR for now. Will re-visit if we see any more BlockedThreads issues in catalog on this path. Thanks all for reviewing !!